stdlib/src/builtin/object.mojo

```diff
@@ -107,20 +120,11 @@ struct _RefCountedListRef:
 
     @always_inline
     fn copy(self) -> Self:
-        _ = self.lst.bitcast[_RefCountedList]()[].refcount.fetch_add(1)
+        _ = self.lst.bitcast[_RefCountedList]()[].impl
         return Self {lst: self.lst}
 
     fn release(self):
-        var ptr = self.lst.bitcast[_RefCountedList]()
-        var prev = ptr[].refcount.fetch_sub(1)
-        if prev != 1:
-            return
-
-        # Run the destructor on the list elements and then destroy the list.
-        var list = __get_address_as_owned_value(ptr.address).impl
-        for i in range(len(list)):
-            list[i].destroy()
-        ptr.free()
+        var ptr = self.lst.bitcast[_RefCountedList]()[].impl
```

Comment:
`__del__` is called after the end of this function, so do `decrement`

Reply:
**Question** Will this also run the destructors for the list elements themselves in the `Arc` destructor, or does it just delete the inner pointer to the list of elements held by `Arc`?

Reply:
I think it calls `__del__`  in Arc, and inner.free() is called. I'm not sure what exactly does free of List do. But I guess it calls `__del__` which `free` the list of elements.

Reply:
I try print out and ensure that inner.free() is called, and so does `__del__`. But there is one weird thing, I see tons of printout in List `__del__`, not sure why does it been called so many times.

Reply:
btw I run the test with

```mojo
from memory._arc import Arc

fn test_123():
    print("== test_123")
    var a = Arc[List[Int32]](List[Int32](1, 2, 3))
    var b = a
    b.set(List[Int32](4, 5, 6))

    b[].append(7)

    # CHECK: 4
    print(a[][0])
    # CHECK: 4
    print(b[][0])
    # CHECK: 4
    print(len(b[]))

fn main():
    test_123()
```

Reply:
The crazy priintouts is also reproduced without arc, so it seems to be another issue, nothing about Arc. If it is unexpected we can file an issue to track it.

```mojo
fn test_456():
    print("== test_456")
    var a = List[Int32](1, 2, 3)
    # CHECK: 1
    print(a[0])
```

Reply:
Interesting that the destructor (even on top-of-tree here, without Arc) seems to be infinitely recursing.  That seems like a compiler bug of some sorts, but I don't see why yet.